Bugfix#75 seonghyeon

castlehyeon opened the pull request:
FIX: 신규신청 시 한번 더 임시저장을 눌렀을 때, 상태관리가 가능하도록, 스웨거 적용시도(dews 충돌로x)
FIX: 시작일과 종료일은 오늘 날짜보다 작을 수 없도록 유효성 처리
FIX:직급, 직위 중 하나만 값이 있을 때, 0126의 경우는 직급 직위가 동일하기 때문에 같은 값을 넣어주는 로직
FIX: 남가연A의 소속이 "//테스트"로 나오는 문제 해결하기
feat: 결재신청 페이지 및 백단 구현 완료
feat: 반려 버튼 기능 구현
FIX: 신규신청 시 임시저장 버튼을 눌렀을 때, 결재요청 버튼이 활성화 되도록
FIX: 결재처리 이후 보여지는 메인페이지에 업데이트 된 상태코드가 보여지도록 수정
FIX: 결제처리 이후 바뀐 상태코드에 따라 버튼 활성화
feat: 결재대기와 결재페이지 구분, 각각 기능 구현
fix: 결재완료일 때는 결재페이지 버튼 비활성화, 결재대기나 결재완료시에만 readonly
feat: 결재요청 취소 구현

castlehyeon committed:
FIX: 신규신청 시 한번 더 임시저장을 눌렀을 때, 상태관리가 가능하도록, 스웨거 적용시도(dews 충돌로x)

- 신규신청 시 메인페이지에서 받은 데이터가 없으므로 별도의 데이터 객체를 만들고서 비교

castlehyeon committed:
FIX: 시작일과 종료일은 오늘 날짜보다 작을 수 없도록 유효성 처리

castlehyeon committed:
FIX:직급, 직위 중 하나만 값이 있을 때, 0126의 경우는 직급 직위가 동일하기 때문에 같은 값을 넣어주는 로직

castlehyeon committed:
FIX: 남가연A의 소속이 "//테스트"로 나오는 문제 해결하기

-상위 부서를 못찾았을 경우 앞의 슬래시 //를 제거한 채로 테이블에 데이터 넣기.

castlehyeon committed:
feat: 결재신청 페이지 및 백단 구현 완료

- 결재요청(후에 결재페이지로 변경할 것) 버튼을 누르면 커스텀 모달창에서 신청 내용을 띄우기.
- 결재 버튼을 누르면 결재상태를 결재완료로 변경

castlehyeon committed:
feat: 반려 버튼 기능 구현

- 반려 버튼 클릭 시 결재상태코드 변경

castlehyeon committed:
FIX: 신규신청 시 임시저장 버튼을 눌렀을 때, 결재요청 버튼이 활성화 되도록

castlehyeon committed:
FIX: 결재처리 이후 보여지는 메인페이지에 업데이트 된 상태코드가 보여지도록 수정

- dialog의 ok API를 사용해서 처리된 modaldata를 메인페이지에 넘겨주도록 수정

castlehyeon committed:
FIX: 결제처리 이후 바뀐 상태코드에 따라 버튼 활성화

crystal9799 commented on castlehyeon/front/ESSODM01400_x10005.html:
```diff
@@ -448,14 +448,22 @@ <h2 id="tit5" class="dews-ui-title">조회정보</h2>
+            if (path && path.startsWith('//')) {
+                path = path.substring(2); // 앞의 두 슬래시를 제거
+            }
+            grid.setCellValue(grid.select(), 'ORG', data[0].ORG === null ? path : data[0].ORG, false); // 소속
```
PATH에 대한 값 처리 수정 확인.

castlehyeon committed:
feat: 결재대기와 결재페이지 구분, 각각 기능 구현

- 신규신청 시 결재페이지 버튼 비활성화
- 임시저장 상태는 결재페이지 버튼 비활성화
- 결재요청 버튼을 눌렀을 시 결재대기로 변경, 결재페이지 활성화

castlehyeon committed:
fix: 결재완료일 때는 결재페이지 버튼 비활성화, 결재대기나 결재완료시에만 readonly

-  반려일 때도 바로 수정가능하다.

castlehyeon committed:
feat: 결재요청 취소 구현

- 결재대기이거나 반려일 경우 결재요청 취소 버튼을 누르면 임시저장으로 바뀌게 구현함.
